components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/internal/service/impl/UserAccountRecoveryManager.java

```diff
@@ -78,6 +87,8 @@ public class UserAccountRecoveryManager {
     private static final boolean PER_USER_FUNCTIONALITY_LOCKING_ENABLED = Utils.isPerUserFunctionalityLockingEnabled();
     private static final String FUNCTIONALITY_PREFIX = "FUNCTIONALITY_";
 
+    AbstractUserStoreManager carbonUM;
```

Comment:
Why do we need to declare a class level variable for this?

Reply:
It is used in getDomainNames method also. We can create it inside the methods also.

Reply:
Resolved by https://github.com/wso2-extensions/identity-governance/pull/623/commits/a1d670e3d1e816b77727e8d627b4caaba19e6c17

Reply:
Resolved by https://github.com/wso2-extensions/identity-governance/pull/623/commits/a1d670e3d1e816b77727e8d627b4caaba19e6c17